src/lib/MapLibre.svelte

```diff
@@ -140,6 +156,29 @@
       dispatch('zoomend', { ...ev, map: $mapInstance });
     });
 
+    // When the basemap style is changed, it nukes all existing layers and sources
+    // Here we listen for style.load events, store the layers and sources that
+    // have come from the new basemap style and then add back in any layers and
+    // styles that where added by the user as sub elements
+
+    $mapInstance.on('style.load', () => {
+      if ($mapInstance) {
+        const mapStyle = $mapInstance.getStyle();
+        lastStyleLayerIds = mapStyle.layers.map((l) => l.id);
+        lastStyleSourceIds = Object.keys(mapStyle.sources);
+        if (sourcesToReAddAfterStyleChange) {
+          for (const [id, source] of Object.entries(sourcesToReAddAfterStyleChange)) {
+            $mapInstance.addSource(id, source);
+          }
+        }
+        if (layersToReAddAfterStyleChange) {
+          for (const layer of layersToReAddAfterStyleChange) {
+            $mapInstance?.addLayer(layer);
```

Comment:
Why's the `?` necessary here, but not above? We're under an `if ($mapInstance)` block; did the flow control check fail?

Reply:
It's not. Going to remove it now.

Reply:
Removed in latest commit